src/components/SemanticModal/SemanticModal.tsx

```diff
@@ -0,0 +1,120 @@
+import * as CustomPropTypes from '../../types/customPropTypes';
+import { getColor, pxToRem } from '../../utils';
+import { Inline, Padbox, Stack } from '../layout';
+import { SpaceSizes } from '../../theme';
+import { SSCIconNames } from '../../theme/icons/icons.enums';
+
+const colorVariants = {
+  // FIXME replace by semantic color success 500
+  [SemanticModalVariants.success]: '#358500',
+  [SemanticModalVariants.warn]: 'pumpkin',
+  [SemanticModalVariants.error]: 'strawberry',
+  [SemanticModalVariants.ban]: 'strawberry',
+};
+
+const iconVariants = {
+  [SemanticModalVariants.success]: SSCIconNames.checkCircle,
+  [SemanticModalVariants.warn]: SSCIconNames.exclTriangle,
+  [SemanticModalVariants.error]: SSCIconNames.error,
+  [SemanticModalVariants.ban]: SSCIconNames.ban,
+};
+
+const CenteredText = styled(Text)`
+  text-align: center;
+`;
+
+const StyledIcon = styled(Icon)<{ $variant?: SemanticModalProps['variant'] }>`
+  color: ${({ $variant }) =>
+    $variant === SemanticModalVariants.success
+      ? colorVariants[$variant]
+      : getColor(colorVariants[$variant])};
+  font-size: ${pxToRem(48)};
+`;
+
+const SemanticModal = forwardRef<HTMLDivElement, SemanticModalProps>(
+  (
+    {
+      title,
+      message,
+      actions,
+      onClose,
+      variant = SemanticModalVariants.success,
+    },
+    ref,
+  ) => {
+    const renderButton = (
+      action: ActionKinds<[React.MouseEvent]>,
+      buttonVariant: typeof ButtonEnums.ButtonVariants[keyof typeof ButtonEnums.ButtonVariants],
+    ) => (
+      <Button
+        key={action.name}
+        href={(action as AbsoluteLinkActionKind<[React.MouseEvent]>).href}
+        name={action.name}
+        size={SpaceSizes.lg}
+        to={(action as RelativeLinkActionKind<[React.MouseEvent]>).to}
+        variant={buttonVariant}
+        onClick={action.onClick}
+      >
+        {action.label}
+      </Button>
+    );
+
+    return (
+      <Modal ref={ref} size={ModalEnums.ModalSizes.sm} onClose={onClose}>
+        <Stack gap={SpaceSizes.md}>
+          <Inline justify="center">
+            <Padbox paddingSize={SpaceSizes.md}>
+              <StyledIcon $variant={variant} name={iconVariants[variant]} />
+            </Padbox>
+          </Inline>
+          <Stack gap={SpaceSizes.md} justify="center">
+            <H3>{title}</H3>
+            <CenteredText>{message}</CenteredText>
+            <Padbox paddingSize={SpaceSizes.md}>
+              <Inline gap={SpaceSizes.md}>
+                {actions.map(
+                  (action: ActionKinds<[React.MouseEvent]>, index: number) =>
+                    isNotUndefined(action) &&
+                    renderButton(
+                      action,
+                      index === 0
+                        ? ButtonEnums.ButtonVariants.outline
+                        : ButtonEnums.ButtonVariants.solid,
+                    ),
+                )}
+              </Inline>
+            </Padbox>
+          </Stack>
+        </Stack>
```

Comment:
This seems to be great ecample when to use `Center` primitive. You can set `isTextCentered` to `true` and get rid of the `CenteredText`. You can also get rid of `justify="center"` on L82 and L87. And add `justify="center"` to L91

Reply:
Updated 😉